### PR TITLE
Use ssh proxy config for tests

### PR DESCRIPTION
**Pull request by timrchavez**

### CODE REVIEW

**Comment by masteinhauser** [comment]
retest

File: test/cleanup [hunk 1]
```diff
@@ -2,4 +2,8 @@
 
 source $(dirname $0)/common
 
-ursula "envs/test" ${ROOT}/playbooks/${TEST_ENV}/tasks/delete.yml
+if [ -f ${ROOT}/envs/test/ssh.config ]; then
+    URSULA_SSH_CONFIG="--ursula-ssh-config=\"${ROOT}/envs/test/ssh.config\""
```
Comment (j2sol) [inline test/cleanup]: ursula-cli already has code to opportunistically lad an ssh config from the environment directory. 
https://github.com/blueboxgroup/ursula-cli/blob/master/ursula_cli/shell.py#L361-L363

Have you tested with that? Might work if you change the file name from `ssh.config` to `ssh_config`.

File: test/setup [hunk 2]
```diff
@@ -79,34 +79,49 @@ if [[ ${TEST_ENV} == "ci-full" ]]; then
...
 eof
+cat > ${ROOT}/envs/test/ssh.config <<eof
+Host jumphost
+    User ubuntu
+    HostName $(public_ip "${testenv_instance_prefix}-controller-0")
+    IdentityFile ${ROOT}/envs/test/${KEY_NAME}.pem
+    ProxyCommand none
+
+Host *
+    ProxyCommand ssh -q -F envs/test/ssh.config jumphost nc -q0 %h %p
```
Comment (j2sol) [inline test/setup]: `ProxyCommand ssh -W %h:%p jumphost`

**Comment by j2sol** [comment]
I like where this is going, just a couple suggestions.

**Comment by timrchavez** [comment]
Testing with the changes you've suggested @j2sol

**Comment by j2sol** [comment]
Hrm, did it work locally? Jenkins didn't like it :(

**Comment by j2sol** [comment]
retest

**Comment by j2sol** [comment]
Please rebase on master.

**Comment by j2sol** [comment]
So this failed testing, ssh connectivity issue :(

**Comment by timrchavez** [comment]
Interesting.  This worked fine for me in my own test.  I'll dig into it.

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
reset

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Comment by timrchavez** [comment]
retest

**Pull request closed without merging**